### PR TITLE
unify KeyPair and Curve interfaces

### PR DESCRIPTION
**Pull request by aricart**
[FIX] unified the nkey and curve key interfaces
[FIX] made CreatePairWithRand delegate to CreateCurveKeysWithRand for curve prefixes [FIX] made FromSeed delegate to FromCurveSeed for curve prefixes

### CODE REVIEW

File: public.go [hunk 1]
```diff
@@ -64,3 +64,14 @@ func (p *pub) Wipe() {
 	p.pre = '0'
 	io.ReadFull(rand.Reader, p.pub)
 }
+
+func (p *pub) Seal(input []byte, recipient string) ([]byte, error) {
+	return nil, ErrCannotSeal
```
Comment (derekcollison) [inline public.go:69]: We might have the private key but its wrong type. I think errors need to be better for these.

File: errors.go [hunk 2]
```diff
@@ -15,28 +15,32 @@ package nkeys
...
+	ErrInvalidRecipient         = nkeysError("nkeys: not a valid recipient public curve key")
+	ErrInvalidSender            = nkeysError("nkeys: not a valid sender public curve key")
+	ErrInvalidCurveKey          = nkeysError("nkeys: not a valid curve key")
+	ErrInvalidCurveSeed         = nkeysError("nkeys: not a valid curve seed")
+	ErrInvalidEncrypted         = nkeysError("nkeys: encrypted input is not valid")
+	ErrInvalidEncVersion        = nkeysError("nkeys: encrypted input wrong version")
+	ErrCouldNotDecrypt          = nkeysError("nkeys: could not decrypt input")
+	ErrInvalidCurveKeyOperation = nkeysError("nkeys: curve key is not valid for sign/verify")
+	ErrInvalidNKeyOperation     = nkeysError("nkeys: only curve key can seal/open")
+	ErrCannotOpen               = nkeysError("nkeys: cannot open no private curve key available")
```
Comment (derekcollison) [inline errors.go:42]: These should not be used when wrong type is being used.

**Review by derekcollison — CHANGES_REQUESTED** [review]

File: public.go (hunk 1 above)
Comment (aricart) [reply public.go:69]: in the case above, we only have a public key - this is the public key version of KeyPair. The message is `nkeys: cannot seal no private curve key available` which aligns with the equivalent one for Sign which is `nkeys: can not sign, no private key available`

**Review by derekcollison — APPROVED** [review]
LGTM!

**Pull request merged**